source/server/active_tcp_socket.h

```diff
@@ -69,6 +77,10 @@ struct ActiveTcpSocket : public Network::ListenerFilterManager,
   // Network::ListenerFilterManager
   void addAcceptFilter(const Network::ListenerFilterMatcherSharedPtr& listener_filter_matcher,
                        Network::ListenerFilterPtr&& filter) override {
+    // update the listener filter max read bytes
+    if (filter->maxReadBytes() > listener_filter_max_read_bytes_) {
+      listener_filter_max_read_bytes_ = filter->maxReadBytes();
```

Comment:
@ggreenway @mattklein123 I think I'm wrong here. Since we can support the filters `proxy protocol+tls inspect` and `proxy protocol + HTTP inspect`, then the maxReadBytes should be the sum of them, not the maximum one, right?

Reply:
I added a new interface `resetCapacity` to the ListenerFilterBuffer, it allows to reset listener filter buffer capacity between filters, then it is avoided to create a buffer whose size is the sum of all the filters' `maxReadBytes()`. It also allows the proxy filter to increase the buffer capacity when it discovers the tlv extensions size, that is avoid creating a maximum header size buffer (maximum to 64kb) in the beginning.

https://github.com/envoyproxy/envoy/pull/17395/commits/af40bae14ea0cec7235cfca7c6e5a097035bc907